promise-native-tool.js

```diff
@@ -173,6 +173,50 @@ class PromiseNativeTool {
 
         return deferred;
     }
+
+    /**
+     * Iterate over all the values in the Iterable into an array and map the array to another using the given mapper function.
+     * Promises returned by the mapper function are awaited for and the returned promise doesn't fulfill until all mapped promises have fulfilled as well.
+     * If any promise in the array is rejected, or any promise returned by the mapper function is rejected, the returned promise is rejected as well.
+     * @param data - an array or object to iterate over
+     * @param iterator - iterator function. Function arguments: (element/value, idx/key)
+     * @param options
+     * @param options.concurrency - concurrency must be >= 1
+     */
+    static map(data, iterator, options) {
```

Comment:
Add `options = { concurrency: 1 }` default.
Also improve jsdoc to set `options` as optional